src/makeCalculator.js

```diff
@@ -4,7 +4,47 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  return {
+    add(value){
+      result += value;
+      return this;
+    },
+
+    subtract(value){
+      result -= value;
+      return this;
+    },
+
+    multiply(value){
+      result *= value;
+      return this;
+    },
+
+    divide(value){
+      if(value !== 0){
+        result /= value;
+        return this;
+      } else {
+        console.log('Error: Division by zero');
+      }
+    },
+
+    reset(){
+      result = 0;
+      return this;
+    },
+
+    operate(operation, value){
+      operation.call(this, value);
+      return this;
```

Comment:
[CODE KNOWLEDGE] - It's good practice to check if the operation parameter is indeed a function before attempting to call it. This can prevent runtime errors if `operate` is used incorrectly. Add a check to ensure that `operation` is a function before invoking it.